### ISSUE TITLE
Build fails with clang-6: error: 'xsgetn' is a protected member of 'std::__1::basic_streambuf<char>'

### ISSUE THREAD

**Issue by yurivict** [ValeevGroup/mpqc#56]
Getting these errors on FreeBSD 12:

```
/usr/local/include -O2 -pipe -fstack-protector -isystem /usr/local/include -fno-strict-aliasing -isystem /usr/local/include -c mstate.cc  -fPIC -DPIC -o .libs/mstate.o
mstate.cc:577:28: error: 'xsgetn' is a protected member of 'std::__1::basic_streambuf<char>'
      *nbuf_buffer = buf_->xsgetn(buffer,bufsize);
                           ^
/usr/include/c++/v1/streambuf:281:24: note: declared protected here
    virtual streamsize xsgetn(char_type* __s, streamsize __n);
                       ^
```

**Comment by yurivict** [comment]
There are also errors like this:
```
disprep.cc:29:44: error: use of undeclared identifier 'endl'; did you mean 'std::endl'?
           << " dispersion coefficient" << endl;
                                           ^~~~
                                           std::endl
```

**Comment by yurivict** [comment]
I'll try the version 3.X and will get back to you.

**Comment by evaleev** [comment]
@yurivict version 3.X is abandoned and will not be maintained. Version 4 is live in master and will be released (as 4.0.0-beta.1) any day now. I'd like to fix 2.3.1 though, if that version is giving you trouble.

**Comment by yurivict** [comment]
2.3.1 builds fine. I may have tried version 3, not sure now.
I will wait for the version 4 then.
Please feel free to close these bug reports then.
